src/NotifyStatusPoller/Runner/JobRunner.php

```diff
@@ -60,13 +61,34 @@ public function run(): void
         }
 
         $this->logger->info('Updating', ['count' => count($inProgressResults), 'context' => Context::NOTIFY_POLLER]);
+
         $updatedCount = 0;
+        $notifyStatusCounts = [
+            NotifyStatus::PENDING_VIRUS_CHECK => 0,
+            NotifyStatus::VIRUS_SCAN_FAILED => 0,
+            NotifyStatus::VALIDATION_FAILED => 0,
+            NotifyStatus::FAILED => 0,
+            NotifyStatus::ACCEPTED => 0,
+            NotifyStatus::RECEIVED => 0,
+            NotifyStatus::CANCELLED => 0,
+            NotifyStatus::TECHNICAL_FAILURE => 0,
+            NotifyStatus::PERMANENT_FAILURE => 0,
+            NotifyStatus::TEMPORARY_FAILURE => 0,
+            NotifyStatus::CREATED => 0,
+            NotifyStatus::SENDING => 0,
+            NotifyStatus::DELIVERED => 0,
+        ];
 
         foreach ($inProgressResults as $getNotifyStatus) {
             try {
                 $updateDocumentStatus = $this->getNotifyStatusHandler->handle($getNotifyStatus);
                 $this->updateDocumentStatusHandler->handle($updateDocumentStatus);
+
                 $updatedCount++;
+
+                if (array_key_exists($updateDocumentStatus->getNotifyStatus(), $notifyStatusCounts)) {
+                    $notifyStatusCounts[$updateDocumentStatus->getNotifyStatus()]++;
+                }
```

Comment:
If you added an `else` to this, you could avoid having to prepopulate all the statuses. That would be more flexible if we added more statuses in the future, and would reduce the amount we're logging (since all the "0"s would be excluded.
```suggestion
                if (array_key_exists($updateDocumentStatus->getNotifyStatus(), $notifyStatusCounts)) {
                    $notifyStatusCounts[$updateDocumentStatus->getNotifyStatus()]++;
                } else {
                    $notifyStatusCounts[$updateDocumentStatus->getNotifyStatus()] = 0;
                }
```

Reply:
Thanks this is done.